vendor/github.com/cilium/ebpf/internal/sys/signals.go

```diff
@@ -0,0 +1,82 @@
+package sys
+
+import (
+	"fmt"
+	"runtime"
+	"unsafe"
+
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+var profSet unix.Sigset_t
+
+func init() {
+	if err := sigsetAdd(&profSet, unix.SIGPROF); err != nil {
+		panic(fmt.Errorf("creating signal set: %w", err))
+	}
+}
```

Comment:
same question here, this looks new and it's unclear what the impact/failure modes could be for binaries that link in this package but don't actually use ebpf at all

Reply:
IMHO: We should reach out to upstream about moving this out of init() / var setup in favor of an explicit `Init()` or similar. I don't think it's reasonable for libraries to automatically register signal handlers regardless.